src/movies.js

```diff
@@ -1,25 +1,68 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors() {}
+
+function getAllDirectors() {
+  let moviedirectors = movies.map(function(e){
+    return e.director
+  })
+return moviedirectors
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies() {}
+function howManyMovies() {
+  let dramaMoviesBySteveSpielberg = movies.filter(function(el){
+    return el.director == 'Steven Spielberg' && el.genre.includes('Drama') == true
+  })
+  return dramaMoviesBySteveSpielberg.length
+}
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage() {}
+function scoresAverage() {
+  let averageScore = movies.map(function(el){
+    return el.score
+
+  })
+}
 
 // Iteration 4: Drama movies - Get the average of Drama Movies
-function dramaMoviesScore() {}
+function scoresAverage() {
+  let averageScore = movies.map(function(el){
+    return el.score})
+  
+  let countofscore = averageScore.length
+  
+  let summedscore = averageScore.reduce(function(a,b){
+    return a+b
+  })
+  return (summedscore/countofscore).toFixed(2)
+  
+}
 
 // Iteration 5: Ordering by year - Order by year, ascending (in growing order)
-function orderByYear() {}
+function orderByYear() {
+  sortedArr = movies.sort(function(a,b){
+    return a.year - b.year
+  })
+  return sortedArr
+}
+
```

Comment:
The sort function mutates the original array, which is not often desired, so instead we should copy the array first with, for instance, with `JSON.parse(JSON.stringify(movies))` or `array.slice()`. Also, in the case of two movies having the same year, we should sort alphabetically by using localeCompare:
```
function orderByYear(array) {
  const sortedArray = array.slice().sort((a, b) => {
    if (a.year !== b.year) {
      return a.year - b.year;
    } else {
      return a.title.localeCompare(b.title);
    }
  });
  return sortedArray;
}
```